site/app/libraries/database/DatabaseQueries.php

```diff
@@ -2256,22 +2256,27 @@ public function getBadGradedComponentsCountByGradingSections($g_id, $sections, $
 
         return $return;
     }
-
-    public function getAverageComponentScores($g_id, $section_key, $is_team, string $bad_submissions, string $null_section) {
+    /**
+     * Gets the number of bad (late) graded submissions associated with this gradeable.
+     *
+     * @param  int $g_id gradeable id we are looking up
+     * @param  string $section_key key we are basing grading sections off of
+     * @param  boolean $is_team true if the gradeable is a team assignment
+     * @param  string $null_section to check if we look in the null section or not
+     * @return int the number of bad submissions
+     */
+    private function getBadSubmissionsCount($g_id, $section_key, $is_team, string $null_section) {
```

Comment:
It would be good to add PHP type annotations here.  PHPDoc annotations aren't verified by anything, and can be incorrect, while native PHP type annotations are checked in development mode.